staging/src/k8s.io/apimachinery/pkg/util/proxy/transport_test.go

```diff
@@ -274,3 +277,65 @@ func TestProxyTransport(t *testing.T) {
 		testItem(name, &item)
 	}
 }
+
+func TestRewriteResponse(t *testing.T) {
+	gzipbuf := bytes.NewBuffer(nil)
+	flatebuf := bytes.NewBuffer(nil)
+	gzw := gzip.NewWriter(gzipbuf)
+	flw, _ := flate.NewWriter(flatebuf, flate.BestCompression)
+	defer gzw.Close()
+	defer flw.Close()
+
+	testTransport := &Transport{
+		Scheme:      "http",
+		Host:        "foo.com",
+		PathPrepend: "/proxy/node/node1:10250",
+	}
+
+	encodeType := []string{"gzip", "deflate"}
+	expectedGzip := "gzip test"
+	expectedDeflate := "deflate test"
+	gzw.Write([]byte(expectedGzip))
+	gzw.Flush()
+	flw.Write([]byte(expectedDeflate))
+	flw.Flush()
+	resp := []http.Response{
```

Comment:
It is better (more understandable, less error prone) to make one table with a struct type than to keep parallel arrays.

Reply:
@lavalamp Thanks!! Updated. PTAL